federator/controllers/amkocluster_controller.go

```diff
@@ -172,6 +171,72 @@ func (r *AMKOClusterReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 	return ctrl.Result{}, nil
 }
 
+func (r *AMKOClusterReconciler) FederateGSLBConfig(ctx context.Context, memberClusters []KubeContextDetails) error {
+	// Determine the state that we need to federate across all member clusters
+	var currGCList gslbalphav1.GSLBConfigList
+	err := r.List(ctx, &currGCList, &client.ListOptions{
+		Namespace: AviSystemNS,
+	})
+	if err != nil {
```

Comment:
if the error is NotFound - should you delete the GSLB objects too in that case from the member clusters?

Reply:
This is a configuration error or a n/w error @sudswasavi. `List()` will return an error if the CRD is not installed, or there's a n/w issue while the call is made. So, we return in such a case and keep on retrying.

"not found" case is not an error and we delete the objects from all member clusters in such a case (handled below):
```
if len(currGCList.Items) == 0 {
		if err = DeleteObjsOnAllMemberClusters(ctx, memberClusters, AviSystemNS, &gslbalphav1.GSLBConfig{}); err != nil {
              [...]
```
In case of config issues or n/w issues, we return the error and keep on retrying.